specification/marketplaceordering/resource-manager/Microsoft.MarketplaceOrdering/stable/2021-01-01/Agreements.json

```diff
@@ -0,0 +1,528 @@
+            "$ref": "./examples/CancelMarketplaceTerms.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PublishersParameter"
+          },
+          {
+            "$ref": "#/parameters/OffersParameter"
+          },
+          {
+            "$ref": "#/parameters/PlansParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request was successfully processed and the terms were rejected.",
+            "schema": {
+              "$ref": "#/definitions/AgreementTerms"
+            }
+          },
+          "default": {
+            "description": "Microsoft.MarketplaceOrdering error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.MarketplaceOrdering/agreements/{publisherId}/offers/{offerId}/plans/{planId}": {
+      "get": {
+        "operationId": "MarketplaceAgreements_GetAgreement",
+        "description": "Get marketplace agreement.",
+        "x-ms-examples": {
+          "SetMarketplaceTerms": {
+            "$ref": "./examples/GetAgreementMarketplaceTerms.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PublishersParameter"
+          },
+          {
+            "$ref": "#/parameters/OffersParameter"
+          },
+          {
+            "$ref": "#/parameters/PlansParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Terms returned successfully",
+            "schema": {
+              "$ref": "#/definitions/AgreementTerms"
+            }
+          },
+          "default": {
+            "description": "Microsoft.MarketplaceOrdering error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.MarketplaceOrdering/agreements": {
+      "get": {
+        "operationId": "MarketplaceAgreements_List",
+        "description": "List marketplace agreements in the subscription.",
+        "x-ms-examples": {
+          "SetMarketplaceTerms": {
+            "$ref": "./examples/ListMarketplaceTerms.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
```

Comment:
Could we please use the subscriptionId and apiversion from common-types instead of defining these by ourselves here?